### DISCUSSION TITLE
Value range for Resonance

### DISCUSSION THREAD

**Discussion by nekton2** [ElectroTechnique/TSynth-Teensy4.1#132] [T4.1 code development]
Is the smallest value of 0.71 OK?  Or is the potentiometer the problem?

Or could the potentiometers be calibrated? Tell the firmware where the potentiometer reached the minimum value and maximum value?

**Comment by wilksy** [comment]
You could just re scale in the code.

**Reply by nekton2** [reply]
Okay, that will probably be a big challenge...

**Comment by wilksy** [comment]
Sorry, presumed as on GitHub you had coding knowledge. 

I can help but going to be couple of weeks until I have time. Are the using the pre -compiled binary? Which version if so

**Reply by nekton2** [reply]
Thanks! I use binary v2.33, but I also have a massive problem with "Background audio noise in v2.33". Adjust the SPI clock as in V2.32? Is the problem solved?

**Comment by nekton2** [comment]
Is the value of 0.71 as the smallest value of the resonance possibly intentional? Unfortunately I'm only a beginner, but in the file 'TSynth.ino', function 'myControlChange" you can find the following source code:

```
case CCfilterres:
    // Pick up
    if (!pickUpActive && pickUp && (resonancePrevValue < ((14.29f * POWER[value - TOLERANCE]) + 0.71f) || resonancePrevValue > ((14.29f * POWER[value + TOLERANCE]) + 0.71f)))
      return; // PICK-UP

    // If <1.1 there is noise at high cutoff freq
    updateFilterRes((14.29f * POWER[value]) + 0.71f);
    resonancePrevValue = (14.29f * POWER[value]) + 0.71f; // PICK-UP
    break;
```

0.71 is added here.

**Reply by Electrotechnique** [reply]
The values for resonance are based on the values used in the code, which start at 0.707. Yes it could start at 0 on the display.

**Reply by nekton2** [reply]
Okay, thanks. Unfortunately, 0.71 is the smallest value here. Can a larger 'safety range' for the potentiometers be set in the source code?

**Comment by nekton2** [comment]
> The values for resonance are based on the values used in the code, which start at 0.707. Yes it could start at 0 on the display.

Okay, what should I do if smallest value is not zero? Perhaps the tolerance of the potentiometer is too great.
